securedrop_client/gui/widgets.py

```diff
@@ -1150,26 +1150,32 @@ def update(self):
                 self.paperclip.hide()
             self.star.update()
         except sqlalchemy.exc.InvalidRequestError as e:
-            logger.error(
-                "Could not update SourceWidget for source %s: %s",
-                self.source.uuid,
-                e
-            )
+            logger.error(f"Could not update SourceWidget for source {self.source_uuid}: {e}")
             raise
 
-    def set_snippet(self, source, uuid=None, content=None):
+    def set_snippet(self, source, uuid=None, content=None) -> None:
         """
         Update the preview snippet only if the new message is for the
         referenced source and there's a source collection. If a uuid and
         content are passed then use these, otherwise default to whatever the
         latest item in the conversation might be.
         """
-        if source == self.source.uuid and self.source.collection:
-            last_collection_object = self.source.collection[-1]
-            if uuid and uuid == last_collection_object.uuid and content:
-                self.preview.setText(content)
-            else:
-                self.preview.setText(str(last_collection_object))
+        if source != self.source_uuid:
+            return
+
+        self.preview.setText("")
+
+        try:
+            self.controller.session.refresh(self.source)
+            if self.source.collection:
+                last_collection_object = self.source.collection[-1]
+                if uuid == last_collection_object.uuid and content:
+                    self.preview.setText(content)
+                else:
+                    self.preview.setText(str(last_collection_object))
+        except sqlalchemy.exc.InvalidRequestError as e:
+            logger.error(f"Could not update snippet for source {self.source_uuid}: {e}")
+            raise
```

Comment:
I don't think we want to raise the error again because it is called in a place of the code that doesn't catch this, see the other `set_snippet` method on the SourceList. Instead if we can't update the snippet because the source was deleted lets say, this could log and fail silently or you could add a try-catch in the other place that calls this, but I think we should minimize all the times we catch just to log and raise again.